### PR TITLE
Lazy probe of GPS

### PR DESCRIPTION
**Pull request by jp-bennett**

### CODE REVIEW

File: src/gps/GPS.cpp [hunk 1]
```diff
@@ -865,6 +865,29 @@ void GPS::publishUpdate()
 
 int32_t GPS::runOnce()
 {
+    if (!GPSInitFinished) {
```
Comment (mverch67) [inline src/gps/GPS.cpp:868]: if (!GPSInitFinished && config.position.gps_enabled) {

File: src/gps/GPS.cpp [hunk 2]
```diff
@@ -1138,24 +1161,24 @@ GPS *createGps()
...
-            new_gps->setup();
-            return new_gps;
-        }
-    } else {
-        GPS *new_gps = new NMEAGPS();
-        new_gps->setup();
-        return new_gps;
-    }
-    return nullptr;
+    /*    if (config.position.gps_enabled) {
```
Comment (mverch67) [inline src/gps/GPS.cpp:1164]: commented code can be removed

File: src/gps/GPS.cpp [hunk 3]
```diff
@@ -1138,24 +1161,24 @@ GPS *createGps()
...
+                // Some boards might have only the TX line from the GPS connected, in that case, we can't configure it at all.
+                // Just assume NMEA at 9600 baud.
+                GPS *new_gps = new NMEAGPS();
+                //new_gps->setup();
+                return new_gps;
+            }
+        } else { */
+    GPS *new_gps = new NMEAGPS();
+    // new_gps->setup();
+    return new_gps;
```
Comment (mverch67) [inline src/gps/GPS.cpp:1180]: return new NMEAGPS();

I save lines of code wherever I can ;-)

**Review by thebentern — APPROVED** [review]
Great code cleanup too!

**Pull request merged**